engine/src/main/java/org/terasology/rendering/world/WorldRendererLwjgl.java

```diff
@@ -193,31 +162,26 @@ public boolean pregenerateChunks() {
     }
 
     @Override
-    public void update(float delta) {
+    public void update(float deltaInMs) {
 
         PerformanceMonitor.startActivity("Update Tick");
```

Comment:
float deltas are in seconds, not milliseconds. So the renamed parameter is incorrect.

I threw performance monitoring around most methods in the renderer to determine where time was spent. Yes, updateTick() probably doesn't have much cost, but I wasn't going to assume that. Feel free to strip some out as you go. In general I guess the approach is to have the monitor around some top level method, and only drilling down if more information is needed.

I put them around methods rather than in methods simply because methods can have multiple return locations - having them around the methods avoid the need for try/finally or multiple endActivity statements, and basically dodges maintenance errors that introduce new return locations.

The Performance monitor is different from profiling. Profiling will tell you how much time is spent in a given method over the life of the application, but we're more interested in how time is spent each frame, and if a frame is unusually slow - why was it slow? If a method takes a second once in the running of an application then that won't really show in the profile statistics - but that would be very noticeable for a user. It also drives the in-game monitoring metrics, which means a user can report where time is spent for them without them having to set up to run from source.